### PR TITLE
TR-13266 Typings for frint

### PR DESCRIPTION
**Pull request by ghost**
**Do**

Add the typings for the different Frint packages.

Since `frint-compat` is maintained for our old applications I consider that it doesn't require the typings.

### CODE REVIEW

**Review by markvincze — APPROVED** [review]

**Comment by markvincze** [comment]
💯

File: packages/frint-react/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import { App } from 'frint';
+import { BehaviorSubject, Observale } from 'rxjs';
```
Comment (fahad19) [inline packages/frint-react/index.d.ts]: `Observable`.

File: packages/frint-react/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+export class Streamer {
+  constructor(...args: any[]);
+
+  set(value: any, ...args: any[]) : Streamer;
```
Comment (fahad19) [inline packages/frint-react/index.d.ts]: any convention for spaces before and after `:`?

File: packages/frint/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+    getParentApps(): App[];
+
+    getProvider(name: string): any;
+
+    getProviders(): any[];
+
+    getRootApp(): App;
+
+    getWidgetInstance(name: string, region?: string, regionKey?: string): App;
```
Comment (fahad19) [inline packages/frint/index.d.ts]: @alexmiranda: makes sense to include deprecated methods here?

Comment (jackTheRipper) [inline packages/frint/index.d.ts]: Since our users can still use it I think yes.

File: packages/frint-react/index.d.ts (hunk 2 above)
Comment (jackTheRipper) [inline packages/frint-react/index.d.ts]: Not really. But we could stick to `whatever: isMyType` for everything

File: packages/frint-react/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+  emit() : void;
+
+  getProps$(): BehaviorSubject<any>;
+
+  getData$(): Observable<any>;
+}
+
+export class Provider extends React.Component {
+  propTypes: React.PropTypes = {
```
Comment (fahad19) [inline packages/frint-react/index.d.ts]: should `React` be imported in this file first?

**Comment by fahad19** [comment]
💖

File: packages/frint-react/index.d.ts (hunk 4 above)
Comment (jackTheRipper) [inline packages/frint-react/index.d.ts]: Yup, I had it in my global scope. Adding it explicitly

**Comment by jackTheRipper** [comment]
Next step will be to create a full example.

For now:

screen shot 2017-04-25 at 16 06 20

And:

screen shot 2017-04-25 at 16 05 51


It looks nice :)

**Review by fahad19 — APPROVED** [review]

**Pull request merged**